src/main/java/org/example/listener/InsultListener.java

```diff
@@ -0,0 +1,65 @@
+package org.example.listener;
+
+import net.dv8tion.jda.api.events.guild.GuildReadyEvent;
+import net.dv8tion.jda.api.events.interaction.command.SlashCommandInteractionEvent;
+import net.dv8tion.jda.api.events.message.MessageReceivedEvent;
+import net.dv8tion.jda.api.hooks.ListenerAdapter;
+import net.dv8tion.jda.api.interactions.commands.build.CommandData;
+import org.example.manager.SlashCommandManager;
+import org.example.service.EventService;
+import org.example.service.InsultService;
+import org.example.slashcommand.SlashCommand;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.Random;
+
+public class InsultListener extends ListenerAdapter {
+    private final EventService eventService;
+    private final InsultService insultService;
+    private final SlashCommandManager slashCommandManager;
+    private final List<SlashCommand> slashCommands;
+    private Integer insultChance = 2;
+
+    public InsultListener() {
+        this.slashCommandManager = SlashCommandManager.getInstance();
+        slashCommands = slashCommandManager.getInsultSlashCommands();
```

Comment:
Are fields like this really needed ? When you need the commands you can get them from manager, I think fields like this are redundant.

Reply:
+ and addressed

Reply:
Due to structure changes those fields are no longer redundant.